tests/unit/test_platform_utils.py

```diff
@@ -150,14 +150,19 @@ def fake_flow():
     db_file = get_workflow_run_pub_db_path(flow_name)
     with CylcWorkflowDAO(db_file, create_tables=True) as dao:
         conn = dao.connect()
-        conn.execute(
-            r"INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
-            r"    VALUES"
-            r"        ('1', 'bar', 1, 'localhost'),"
-            r"        ('1', 'baz', 1, 'dairy'),"
-            r"        ('1', 'bar', 2, 'dairy'),"
-            r"        ('2', 'baz', 1, 'milk')"
-        )
+        conn.execute(r"""
+            INSERT INTO task_jobs (
+                cycle, name, submit_num, platform_name, submit_status
+            )
+                VALUES
+                    ('1', 'bar', 1, 'localhost', 0),
```

Comment:
I think that this test should also include a case where some but not all submissions to a platform have failed:

```suggestion
                    ('1', 'bar', 1, 'localhost', 0),
                    ('1', 'bar', 1, 'turning', 0),
                    ('1', 'bar', 2, 'turning', 1),
```